xbmc/music/MusicDatabase.cpp

```diff
@@ -3823,20 +3817,36 @@ bool CMusicDatabase::GetSongsFullByWhere(const std::string &baseDir, const Filte
 
     // Apply the limiting directly here if there's no special sorting but limiting
     bool limited = extFilter.limit.empty() && sortDescription.sortBy == SortByNone &&
-                   (sortDescription.limitStart > 0 || sortDescription.limitEnd > 0);
+      (sortDescription.limitStart > 0 || sortDescription.limitEnd > 0);
     if (limited)
       strSQLExtra += DatabaseUtils::BuildLimitClause(sortDescription.limitEnd, sortDescription.limitStart);
 
+    std::string strSQL;
     if (!artistData)
-      strSQL = PrepareSQL(strSQL, !filter.fields.empty() && filter.fields.compare("*") != 0 ? filter.fields.c_str() : "songview.* ") + strSQLExtra;
+      strSQL = "SELECT songview.* FROM songview " + strSQLExtra;
     else
-    {
-      strSQL = PrepareSQL(strSQL, !filter.fields.empty() && filter.fields.compare("*") != 0 ? filter.fields.c_str() : "songview.*, songartistview.* ");
+    { // Get data from song and song_artist tables to fully populate songs with artists
```

Comment:
Another minor.. Why not use else if? Imho it's nicer.. Or if you prefer the nested if avoid the negation and do
```
if (artistData)
{
  if (limited)
    strSQL = "foo1";
  else
    strSQL = "foo2";
}
else
  strSQL = "foo3";
```